### PR TITLE
Revision tree tests

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Test cases for revision trees.
Also contains a bugfix for an existing replication test.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This PR adds tests that were used to verify all changes:

### CODE REVIEW

File: lib/Containers/MerkleTree.cpp [hunk 1]
```diff
@@ -628,6 +632,32 @@ void MerkleTree<Hasher, BranchingBits>::checkConsistency() const {
   checkInternalConsistency();
 }
 
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+// intentionally corrupts the tree. used for testing only
+template <typename Hasher, std::uint64_t const BranchingBits>
+void MerkleTree<Hasher, BranchingBits>::corrupt(std::uint64_t count, std::uint64_t hash) {
```
Comment (neunhoef) [inline lib/Containers/MerkleTree.cpp:638]: Cool function!

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by KVS85 — COMMENTED** [review]
LGTM.

**Review by gnusi — COMMENTED** [review]
suggest a micro-optimization

File: arangod/RocksDBEngine/RocksDBMetaCollection.cpp [hunk 2]
```diff
@@ -673,19 +692,50 @@ void RocksDBMetaCollection::revisionTreeSummary(VPackBuilder& builder, bool from
...
+      // no revision tree yet
+      return;
+    }
+  }
+  
+  std::unique_lock<std::mutex> guard(_revisionBufferLock);
+
+  uint64_t inserts = _revisionInsertBuffers.size();
+  uint64_t removes = _revisionRemovalBuffers.size();
+  uint64_t truncates = _revisionTruncateBuffer.size();
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBMetaCollection.cpp]: Ok, done.

**Review by gnusi — APPROVED** [review]

**Pull request merged**